packages/dev/core/src/AudioV2/abstractAudio/subNodes/abstractAudioSubGraph.ts

```diff
@@ -21,31 +21,19 @@ import type { AudioSubNode } from "./audioSubNode";
 export abstract class _AbstractAudioSubGraph {
     private _createSubNodePromises: { [key: string]: Promise<_AbstractAudioSubNode> } = {};
     private _isDisposed = false;
-    private _subNodes: { [key: string]: AbstractNamedAudioNode } = {};
+    private _subNodes: { [key: string]: _AbstractAudioSubNode } = {};
 
     /**
      * Executes the given callback with the named sub node, creating the sub node if needed.
      *
-     * Note that `callback` is executed synchronously if the sub node already exists, otherwise the sub node is created
-     * asynchronously before `callback` is executed.
-     *
      * @param name The name of the sub node
      * @param callback The function to call with the named sub node
      *
      * @internal
      */
-    public callOnSubNode<T extends _AbstractAudioSubNode>(name: AudioSubNode, callback: (node: T) => void): void {
-        const node = this.getSubNode(name);
-        if (node) {
-            callback(node as T);
-            return;
-        }
-
-        const promise = this._createSubNodePromises[name] ?? this.createAndAddSubNode(name);
-
-        promise.then((node) => {
-            callback(node as T);
-        });
+    public async callOnSubNode<T extends _AbstractAudioSubNode>(name: AudioSubNode, callback: (node: T) => void): Promise<void> {
+        await this._createSubNodePromisesResolved();
+        callback(this.getSubNode(name) || ((await this.createAndAddSubNode(name)) as T));
```

Comment:
I don't think I want this to be async. This will cause property setters to always be set asynchronously.

Reply:
Do you mean removing the async keyword, or making everything synchronous?

Reply:
I want this to execute `callback` synchronously if the node exists, otherwise it should execute `callback` after the subnode is created, which is asynchronous.

Reply:
I'm thinking this ...

```TypeScript
    public callOnSubNode<T extends _AbstractAudioSubNode>(name: AudioSubNode, callback: (node: T) => void): void {
        const node = this.getSubNode(name);
        if (node) {
            callback(node as T);
            return;
        }

        this._createSubNodePromisesResolved().then(() => {
            const node = this.getSubNode(name);
            if (node) {
                callback(node as T);
                return;
            }

            this.createAndAddSubNode(name).then((node) => {
                callback(node as T);
            });
        });
    }
```

Reply:
> I'm thinking this ...

This works, and I'm not seeing a way to make it more robust or cleaner. Do you see any improvements that can be made here?

Reply:
no, that looks good to me

Reply:
Ok, thanks. This PR should be good to go then.